src/include/sof/schedule/dp_schedule.h

```diff
@@ -0,0 +1,99 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/*
+ * Copyright(c) 2023 Intel Corporation. All rights reserved.
+ *
+ * Author: Marcin Szkudlinski
+ */
+
+#ifndef __SOF_SCHEDULE_DP_SCHEDULE_H__
+#define __SOF_SCHEDULE_DP_SCHEDULE_H__
+
+#include <sof/schedule/task.h>
+#include <sof/trace/trace.h>
+#include <user/trace.h>
+#include <stdint.h>
+
+/**
+ *
+ * DP scheduler is a scheduler that creates a separate preemptible Zephyr thread for each SOF task
+ *   There's only one instance of DP in the system, however, the threads may run on any core
+ *   Threads are pinned to specific core, there's no SMP processing.
+ *
+ * The task execution may be delayed and task may be re-scheduler periodically
+ * NOTE: delayed start and rescheduling takes place in sync with LL scheduler, meaning the
+ *   DP scheduler is triggered as the last task of LL. That implicates a limitation: LL scheduler
+ *   MUST be running on primary core in order to have this feature working.
```

Comment:
given we currently run a LL scheduler on each core, might be easier to express this that the DP scheduler is triggered by the last task of the LL running on primary core..?

Reply:
"ask of the LL running on primary core" sounds ok,
I just need to make it more clear - we CAN have LL running on every core, but we DON'T have one till a pipeline is running on the core in question. Theoretically, one can start an LL pipeline on core 1, leaving core 0 idle, can't he? In this case DP won't work properly.